sdk/src/wallet/account/operations/output_claiming.rs

```diff
@@ -50,88 +51,55 @@ impl Account {
-                                    &account_details.addresses_with_unspent_outputs,
-                                    // outputs controlled by an alias or nft are currently not considered
-                                    &[],
-                                    output,
-                                    local_time,
-                                    // Not relevant without alias addresses
-                                    None,
-                                )?
-                            {
-                                match outputs_to_claim {
-                                    OutputsToClaim::MicroTransactions => {
-                                        if let Some(sdr) = basic_output.unlock_conditions().storage_deposit_return() {
-                                            // Only micro transaction if not the same
-                                            if sdr.amount() != basic_output.amount() {
-                                                output_ids_to_claim.insert(output_data.output_id);
-                                            }
-                                        }
-                                    }
-                                    OutputsToClaim::NativeTokens => {
-                                        if !basic_output.native_tokens().is_empty() {
-                                            output_ids_to_claim.insert(output_data.output_id);
-                                        }
-                                    }
-                                    OutputsToClaim::All => {
+            if !account_details.locked_outputs.contains(output_id) && account_details.outputs.contains_key(output_id) {
+                if let Some(unlock_conditions) = output_data.output.unlock_conditions() {
+                    // If there is a single [UnlockCondition], then it's an
+                    // [AddressUnlockCondition] and we own it already without
+                    // further restrictions
+                    if unlock_conditions.len() != 1
+                        && can_output_be_unlocked_now(
+                            // We use the addresses with unspent outputs, because other addresses of the
+                            // account without unspent outputs can't be related to this output
+                            &account_details.addresses_with_unspent_outputs,
+                            // outputs controlled by an alias or nft are currently not considered
+                            &[],
+                            output_data,
+                            local_time,
+                            // Not relevant without alias addresses
+                            None,
+                        )?
+                    {
+                        match outputs_to_claim {
+                            OutputsToClaim::MicroTransactions => {
+                                if let Some(sdr) = unlock_conditions.storage_deposit_return() {
```

Comment:
I know, not the part you added, but what do you think about adding this here?
```suggestion

                                if let Some(sdr) = unlock_conditions.storage_deposit_return() {
                                    // If expired, it's not a micro transaction anymore
                                    if unlock_conditions.is_expired(local_time) {
                                        continue;
                                    }
```